Makefile

```diff
@@ -12,30 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-test: lint
+ENV = $(CURDIR)/env
+PYTHON3 = $(ENV)/bin/python3
+
+test: $(ENV) lint
 	@python3 -m unittest
 
 lint: quotelint check-copyright
-	@autopep8 -r --aggressive --diff --exit-code .
-	@flake8 --config=.flake8
+	${PYTHON3} -m autopep8 -r --aggressive --diff --exit-code --exclude env .
+	${PYTHON3} -m flake8 --config=.flake8 --exclude env .
 
 quotelint:
-	@x=$$(grep -rnH --include \*.py "\\\\[\"']" .);                       \
+	@x=$$(grep -rnH --include \*.py "\\\\[\"']" --exclude-dir env .);                       \
 	if [ "$$x" ]; then                                                    \
 		echo "Please fix the quoting to avoid spurious backslashes:"; \
 		echo "$$x";                                                   \
 		exit 1;                                                       \
 	fi >&2
 
 check-copyright:
-	@x=$$(find . -name \*.py -not -empty -type f -print0 | xargs -0 grep -L "^# Copyright"); \
+	@x=$$(find . -name env -prune -o -name \*.py -not -empty -type f -print0 | xargs -0 grep -L "^# Copyright"); \
 	if [ "$$x" ]; then                                                                       \
 		echo "Please add copyright headers to the following files:";                     \
 		echo "$$x";                                                                      \
 		exit 1;                                                                          \
 	fi >&2
 
+$(ENV):
+	python3 -m venv $(ENV)
+	${PYTHON3} -m pip install -r requirements-test.txt
```

Comment:
is that a common spelling? charms.reactive used "test_requirements.txt".
I don't have much stake in it, just don't want it to be surprising for people.

Reply:
We just talked about this with John. The nice thing about start with "requirem.." is that both files are grouped to the eye when searching for them... Like, a developer searching if the project has requirement file(s), their will find both at once.

Reply:
That said, in other projects I used `requirements-dev.txt`

Reply:
a search on github finds an equal number (~30k) of `test_requirements.txt`, `test-requirements.txt`, `requirements_test.txt`, and `requirements-test.txt`, FWIW.